File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_nothink/20260910_092923/python/008_prime_numbers/solution.py
import sys

def is_prime(num):
    """Checks if a number is prime."""
    if num < 2:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False
    # Check odd numbers up to the square root of num
    for i in range(3, int(num**0.5) + 1, 2):
        if num % i == 0:
            return False
    return True

def main():
    # Ensure an argument is provided
    if len(sys.argv) != 2:
        return

    try:
        # Convert the command line argument to an integer
        n = int(sys.argv[1])
    except ValueError:
        # If the argument is not an integer, the program exits silently
        return

    # Iterate through all numbers from 2 up to and including N
    for i in range(2, n + 1):
        if is_prime(i):
            print(i)

if __name__ == "__main__":
    main()

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_nothink/20260910_092923/typescript/008_prime_numbers/solution.ts
/**
 * This program accepts a single integer N from the command line
 * and prints all prime numbers up to and including N.
 */

function isPrime(num: number): boolean {
    if (num <= 1) return false;
    if (num <= 3) return true;
    if (num % 2 === 0 || num % 3 === 0) return false;

    // Optimization: Check divisors up to the square root of num
    // All primes are of the form 6k ± 1
    for (let i = 5; i * i <= num; i += 6) {
        if (num % i === 0 || num % (i + 2) === 0) {
            return false;
        }
    }
    return true;
}

function main() {
    // process.argv[0] is node executable, process.argv[1] is the script path
    const arg = process.argv[2];

    if (arg === undefined) {
        return;
    }

    const n = parseInt(arg, 10);

    // If the argument is not a valid number, do nothing
    if (isNaN(n)) {
        return;
    }

    // Iterate from 2 up to N and print if the number is prime
    for (let i = 2; i <= n; i++) {
        if (isPrime(i)) {
            process.stdout.write(i.toString() + '\n');
        }
    }
}

main();